kmod/src/inode.c

```diff
@@ -534,10 +565,16 @@ int scoutfs_setattr(struct dentry *dentry, struct iattr *attr)
 	if (ret)
 		goto out;
 
+	if (scoutfs_inode_worm_denied(inode)) {
```

Comment:
This needs to go much higher, before the inode is modified.   Just after inode_change_ok makes sense.  And then it doesn't need to mess around with the cleanup labels.